app/controllers/spree/admin/product_catalogs_controller.rb

```diff
@@ -17,10 +18,13 @@ def collection
       end
 
       def load_data
-        @variants = Spree::Product.all.map(&:variants_including_master).flatten
         @product_catalogs = Spree::ProductCatalog.where(store: current_store).order(:name)
       end
 
+      def load_variants
+        @variants = Spree::Product.all
```

Comment:
Uhm, not sure about this, if the store has thousands of products it'd be problematic, is there any way to avoid calling `all` ? maybe limiting products by store or something

Reply:
+ 1 to being more semantic

Reply:
I tried to track this change back to its source and intent, and it looks like originally it was intended to link all Variants to a ProductCatalog (the idea being, by default you want everything in the catalog). That was accomplished by calling `variants_including_master` which is not the right thing to do, as master variants on a product with other variants are not sellable.

If the intent is the same here, then we probably need to agree that sending a special boolean "all products" will put all sellable variants into the product catalog being created. That seems safer than loading every product / variant object into memory.

If that's not possible, we need to go ahead and create a custom array for the select element that contains variant IDs and probably SKUs. We can still avoid paying the full penalty for object creation if we know exactly what the view requires.

Reply:
I think that what I described may have been done already in the other PR:

https://github.com/geminimvp/engine_storefront/pull/455/files#diff-117ea643f27a672f43375fd2348db2d4R6

Reply:
I was expecting to use `Spree::Product.all.map(&:variants_with_master)`, I made change here https://github.com/geminimvp/solidus_product_feed/pull/11/files#diff-154536b14bceed66fa7c2eec20e60704R25, let me know what do you think

Reply:
If this has to be a table of all variants in the store, it needs to be paginated. Right now it would generate a large page for stores with a decent number of products / variants.